server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -421,6 +423,11 @@ private IndexCommit getStartingCommitPoint() throws IOException {
                 // Use the last commit
                 existingCommits = DirectoryReader.listCommits(store.directory());
                 startingIndexCommit = existingCommits.get(existingCommits.size() - 1);
+                if (logger.isDebugEnabled()) {
```

Comment:
we will already log the safe commits and deletions in the onInit of the CombinedDeletionPolicy, maybe this is an overkill?